src/server/services/registrar.js

```diff
@@ -164,8 +164,61 @@ async function getShortname(offeredTermCode, classSectionID) {
   }
 }
 
+/**
+ * Gets the week number, given an academic term and a date
+ *
+ * @param {string} term Academic term, formatted as YYQ (e.g. 20S)
+ * @param {string} date Date string formatted as MM/DD/YYYY
+ * @returns {?string} Returns week number if date is valid in term
+ */
+async function getWeekNumber(term, date) {
+  try {
+    const response = await registrar.call({
+      url: `sis/api/v1/Dictionary/TermSessionsByWeek?SessionTermCode=${term}&SessionCode=RG&PageSize=12`,
+    });
+    if (response === null) return null;
+    const {
+      termSessionsByWeek: [{ termSessionCollection: weeksArray }],
+    } = response;
+
+    // Splits the date parameter into an array: [0] MM, [1] DD, [2] YYYY
+    const dateElements = date.split('/');
+    const dateToCheck = new Date(
+      dateElements[2], // Year
+      dateElements[0] - 1, // Month - 1, because the month is 0-indexed
+      dateElements[1] // Date
+    );
+
+    for (const week of weeksArray) {
```

Comment:
@chern This week handling should have a test. If you have caching working then you can inject some fake data into the cache and then call this function. It should let you test the logic of the code.

Reply:
@chern I have an example of how I mocked a Registrar call here if that helps: https://github.com/ucla/lti-course-requestor/commit/db6eba08100d07e1a5839a977255ed32c0413465#diff-7390adebf90453bf3bda13508bed773eR5